www/js/app.js

```diff
@@ -35,31 +35,37 @@ angular
 
         // Handle the android's hardware button
         $ionicPlatform.registerBackButtonAction(function(){
+          if(appStateService.isPopupOpen){
+                    return;
+          }
+          else if(!appStateService.isPopupOpen){
             if($state.current.name === "options"){
-                alertService.confirmExit(function(res){
-                    if(res == 1){
-                        sessionStorage.removeItem('start');
-                        sessionStorage.removeItem('_interval');
-                        navigator.app.exitApp();
-                    }
-                });
-            }
-            else if($state.current.name === "signin" || $state.current.name === "home" ){
-                navigator.app.exitApp();
-            }
-            else if($state.current.name === "trade" && appStateService.purchaseMode){
-                return;
-            }
-            else if($state.current.name === "trade" && !appStateService.purchaseMode && !appStateService.tradeMode){
-                appStateService.tradeMode = true;
-                if(!$rootScope.$$phase){
-                    $rootScope.$apply();
-                }
-            }
-            else{
-                navigator.app.backHistory();
-            }
-        }, 100);
+                  alertService.confirmExit(function(res){
+                      if(res == 1){
+                          sessionStorage.removeItem('start');
+                          sessionStorage.removeItem('_interval');
+                          navigator.app.exitApp();
+                      }
+                  });
+              }
+              else if($state.current.name === "signin" || $state.current.name === "home" ){
+                  navigator.app.exitApp();
+              }
+              else if($state.current.name === "trade" && appStateService.purchaseMode){
+                  return;
+              }
+              else if($state.current.name === "trade" && !appStateService.purchaseMode && !appStateService.tradeMode){
+                  appStateService.tradeMode = true;
+                  if(!$rootScope.$$phase){
+                      $rootScope.$apply();
+                  }
+              }
+              else{
+                  navigator.app.backHistory();
+              }
+          }
+
+        }, 401);
```

Comment:
why 401 ? following to the angular reference the priority values are as below:
```The priorities for the existing back button hooks are as follows: Return to previous view = 100 Close side menu = 150 Dismiss modal = 200 Close action sheet = 300 Dismiss popup = 400 Dismiss loading overlay = 500```
http://ionicframework.com/docs/api/service/$ionicPlatform/